src/utils/validation.ts

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Validates the given value as a bigint.
+ *
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @throws - Will throw if not a valid bigint.
+ */
+export function validateBigint(name: string, value: unknown, valueKind: string): void {
+  if (typeof value !== "bigint") {
+    throwValidationError(name, value, valueKind, "type 'bigint'");
+  }
+}
+
+/**
+ * Validates the given value as an object.
+ *
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @throws - Will throw if not a valid object.
+ */
+export function validateObject(name: string, value: unknown, valueKind: string): void {
+  if (typeof value !== "object") {
+    throwValidationError(name, value, valueKind, "type 'object'");
+  }
+  if (value === null) {
+    throwValidationError(name, value, valueKind, "non-null");
+  }
+}
+
+/**
+ * Validates the given value as an optional object.
+ *
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @param model - A model object that contains all possible fields. 'value' does not need to have all fields, but it may not have any fields not contained in 'model'.
+ * @throws - Will throw if not a valid optional object.
+ */
+export function validateOptionalObject(
+  name: string,
+  value: unknown,
+  valueKind: string,
+  model: Record<string, unknown>
+): void {
+  if (!value) {
+    // This is okay, the object is optional.
+    return;
+  }
+  validateObject(name, value, valueKind);
+
+  // Check if all given properties of value also exist in the model.
+  for (const property in value as Record<string, unknown>) {
+    if (!(property in model)) {
+      throw new Error(`Object ${valueKind} '${name}' contains unexpected property '${property}'`);
+    }
+  }
+}
+
+/**
+ * Validates the given value as a number.
+ *
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @throws - Will throw if not a valid number.
+ */
+export function validateNumber(name: string, value: unknown, valueKind: string): void {
+  if (typeof value !== "number") {
+    throwValidationError(name, value, valueKind, "type 'number'");
+  }
+}
+
+/**
+ * Validates the given value as a string.
+ *
+ * @param name - The name of the value.
+ * @param value - The actual value.
+ * @param valueKind - The kind of value that is being checked (e.g. "parameter", "response field", etc.)
+ * @throws - Will throw if not a valid string.
+ */
+export function validateString(name: string, value: unknown, valueKind: string): void {
```

Comment:
maybe we could use convenience method `validateParam(name: string, value: unknown)` seeing as 85% of all cases are `validateString("paramname", paramvalue, "parameter")`

Reply:
Well there's different input types apart from just "string" and we'd need a convenience method for all of them :P I do agree it's pretty repetetive, idk